cdap-proto/src/main/java/co/cask/cdap/proto/WorkflowNodeThrowable.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.proto;
+
+import co.cask.cdap.proto.codec.WorkflowNodeThrowableCodec;
+
+import javax.annotation.Nullable;
+
+/**
+ * Carries {@link Throwable} information in {@link WorkflowNodeStateDetail}.
+ */
+public final class WorkflowNodeThrowable {
+  private final String className;
+  private final String message;
+  private final StackTraceElement[] stackTraces;
+  private final WorkflowNodeThrowable cause;
+
+  /**
+   * This constructor is used by {@link WorkflowNodeThrowableCodec} during de-serialization.
+   *
+   * @param className the name of the Throwable
+   * @param message the message inside the Throwable
+   * @param stackTraces stack traces associated with the Throwable
+   * @param cause cause associated with the Throwable
+   */
+  public WorkflowNodeThrowable(String className, String message, StackTraceElement[] stackTraces,
```

Comment:
@chtyim When we create an instance of `WorkflowNodeThrowable` from `Throwable`, we store className, message and stack trace associated with the `Throwable` in member variables.

Now `Throwable` might also have another `Throwable` as a cause. Information about the cause is stored in `WorkflowNodeThrowable cause`. 

So stacktrace need to be separate, right?